src/components/application_manager/test/resumption/resume_ctrl_test.cc

```diff
@@ -705,6 +728,9 @@ TEST_F(ResumeCtrlTest, SetAppHMIState_HMIFull_WithPolicy_DevDisallowed) {
 TEST_F(ResumeCtrlTest, SaveAllApplications) {
   utils::SharedPtr<application_manager::Application> app_sh_mock =
       ::utils::MakeShared<application_manager_test::MockApplication>();
+  EXPECT_CALL(*static_cast<application_manager_test::MockApplication*>(
```

Comment:
@vlantonov Too complicated EXPECT_CALL. Why did you create `SharedPtr<Application>` then cast to `MockApplication*`? May be it's worth to create `SharedPtr<MockApplication>` and make EXPECT_CALL more simple.